### PR TITLE
[image_tools] Use ROS parameters instead of regular CLI arguments

### PR DESCRIPTION
**Pull request by jacobperron**
Using of ROS parameters make the code a little tidier, though a little more verbose on the command-line.
IMO, it seems like a good way to demonstrate best practices as well.

Example usage:

```sh
$ ros2 run image_tools cam2image --ros-args -p burger_mode:=true -p reliability:=reliable
```
```sh
$ ros2 run image_tools showimage --ros-args -p reliability:=reliable
```

Connects to #390

### CODE REVIEW

**Comment by dirk-thomas** [comment]
How does a user get a similar help than what was provided by `--help` before?

**Comment by jacobperron** [comment]
> How does a user get a similar help than what was provided by --help before?

Good question, I guess there's no equivalent to `--help`.

Maybe this is a good example for a feature/tool that is lacking for ROS parameters.
At the moment, you can list parameters for the node, but only after it is running:

```sh
$ ros2 param list
/showimage:
  depth
  history
  reliability
  show_image
  use_sim_time
```

In addition it would be nice if we could see the types/constraints on these params (https://github.com/ros2/ros2cli/issues/362).

And even better (equivalent to `--help`) would be to introspect the node before actually running it. I imagine some like

```sh
$ ros2 run image_tools showimage --ros-args --help
Parameters:
    <list of parameters and descriptors>
Publishers:
    <list of publisher types and topics>
Subscriptions:
    <list of subscription types and topics>
... etc for services and actions.
```

I suppose this kind of introspection feature needs a lot more thought and discussion to make it happen.

**Comment by dirk-thomas** [comment]
Not relevant in this case where the node is implemented in C++. But in the case of Python nodes using `argparse` another major advantage is getting free `argcomplete` functionality. So switching a Python node to parameters would loose that too which I think would be undesired.

**Comment by jacobperron** [comment]
I've added back an option for help (`-h` / `--help`): da736d0

**Comment by jacobperron** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/8415/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/4326/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/6850/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/8314/)

File: image_tools/src/cam2image.cpp [hunk 1]
```diff
@@ -118,55 +96,151 @@ class Cam2Image : public rclcpp::Node
...
+  }
+
+  IMAGE_TOOLS_LOCAL
+  bool help(const std::vector<std::string> args)
+  {
+    if (std::find(args.begin(), args.end(), "--help") != args.end() ||
+      std::find(args.begin(), args.end(), "-h") != args.end())
+    {
+      std::stringstream ss;
+      ss << "Usage: cam2image [-h] [--ros-args [-p param:=value] ...]" << std::endl;
```
Comment (dirk-thomas) [inline image_tools/src/cam2image.cpp]: Nitpick: each `std::endl` also flushes the stream which doesn't seem necessary on each line.

**Comment by jacobperron** [comment]
> I was hoping the nodes would declare their parameters and --help would then call a generic function which get those descriptors and renders the usage.

It seemed easier to do something like was done before to ensure formatting looked okay.
But, this is a nice idea. I think that a generic function like, `get_node_usage()` should live in the client library or a utility somewhere that node developers can use. I can open up a feature request for this.

**Pull request merged**